### PR TITLE
Adding a first draft of `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by kallewesterling**
- [x] Final edits needed

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# DeepSensor: Contributing Guidelines
+
+🎉🎈🍰 Welcome to DeepSensor repository! 🍰🎈🎉
+
+💫🐥 ☀️We're excited that you're here and want to contribute.☀️ 🐥💫
+
+Thank you for considering a contribution to the DeepSensor project! DeepSensor is a Python package dedicated to modeling environmental data with neural processes. With DeepSensor, we aim to provide a user-friendly, effective, and robust solution for the environmental research community. Our vision is <VISION>. By joining our efforts, you're helping push the boundaries of environmental science and machine learning, making strides towards a clearer understanding of our world.
```
Comment (kallewesterling) [inline CONTRIBUTING.md:7]: @tom-andersson We had been discussing a vision statement for DeepSensor. I can remove it for now, unless you have anything prepared or even drafted that we can slot in here?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+We want to ensure that every user and contributor feels welcome, included and supported to participate in DeepSensor community. Whether you're a seasoned developer, a machine learning enthusiast, an environmental scientist, or just someone eager to learn and contribute, **you are welcome here**. We value every contribution, be it big or small, and we appreciate the unique perspectives you bring to the project.
+
+We hope that the information provided in this document will make it as easy as possible for you to get involved. If you find that you have questions that are not discussed below, please let us know through one of the many ways to [get in touch](#get-in-touch).
+
+## Inclusivity
+
+We aim to make DeepSensor a collaboratively developed project. We, therefore, require that all our members and their contributions **adhere to our Code of Conduct**. Please familiarize yourself with our Code of Conduct that lists the expected behaviours.
```
Comment (kallewesterling) [inline CONTRIBUTING.md:15]: - [ ] Kalle to add: a CODE_OF_CONDUCT.md file, see TTW for a good example.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+Every contributor is expected to adhere to our Code of Conduct. It outlines our expectations and ensures a safe, respectful environment for everyone.
+
+## Important Resources
+
+If you'd like to find out more about DeepSensor, make sure to check out:
+
+1. **README**: For a project overview, setup, and usage, please refer to our README.
+2. **Project Roadmap**: Familiarize yourself with our direction and goals by checking out the project's roadmap.
```
Comment (kallewesterling) [inline CONTRIBUTING.md:24]: - [ ] @tom-andersson to send to Kalle his draft of a roadmap, for us to start working on.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+## Get in touch
+
+The easiest way to get involved with the active development of DeepSensor is to join our regular community calls. <more info about community calls here>
```
Comment (kallewesterling) [inline CONTRIBUTING.md:28]: @tom-andersson what do you think we should say here? That the community calls are currently on a hiatus, but that, should the reader be interested, they should join the Slack workspace, where conversation about when to hold the community calls in the future will take place?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+**Slack Workspace**: Join our DeepSensor Slack channel for discussions, queries, and community interactions.
+
+**Email**: If you prefer a more formal communication method or have specific concerns, please reach us at <ADD EMAIL ADDRESS>.
```
Comment (kallewesterling) [inline CONTRIBUTING.md:32]: I'll add my email address in here, and also change the one in the documentation, for now? Unless you want to change it to your personal email, @tom-andersson ?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+If you want to follow along with project developments, you can do so in a number of different ways:
+
+- <DeepSensor on Twitter -- do we want that?>
```
Comment (kallewesterling) [inline CONTRIBUTING.md:36]: Nah. Let's drop this one, if you're ok with that @tom-andersson ?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+- Subscribe to our newsletter <should we have that?>
```
Comment (kallewesterling) [inline CONTRIBUTING.md:37]: Let's pause this as well, and maybe add this to a roadmap instead.

File: CONTRIBUTING.md (hunk 6 above)
Comment (kallewesterling) [reply CONTRIBUTING.md:36]: (We could feasibly start a bluesky account if we wanted to, but even that seems a bit premature perhaps?)

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+- A clear title.
+- A detailed description of the bug.
+- Steps to reproduce it.
+- Expected versus actual behavior.
+
+## Recognising Contributions
+
+We value and recognize every contribution. All contributors will be acknowledged in our CONTRIBUTORS.md file. Notable contributions will also be highlighted in our regular community updates.
+
+<we could also consider adding all-contributors, and add the following section from TTW>
```
Comment (kallewesterling) [inline CONTRIBUTING.md:70]: Remove this comment and add the following section! :) :) :)

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+**PLEASE NOTE: Only one contributor can be added with the bot at a time!** Add each contributor in turn, merge the pull request and delete the branch (`all-contributors/add-<username>`) before adding another one. Otherwise, you can end up with dreaded [merge conflicts](https://help.github.com/articles/about-merge-conflicts). Therefore, please check the open pull requests first to make sure there aren't any [open requests from the bot](https://github.com/tom-andersson/deepsensor/pulls/app%2Fallcontributors) before adding another.
+
+What happens if you accidentally run the bot before the previous run was merged and you got those pesky merge conflicts? (Don't feel bad, we have all done it! 🙈) Simply close the pull request and delete the branch (`all-contributors/add-<username>`). If you are unable to do this for any reason, please let us know on Slack <link to Slack> or by opening an issue, and one of our core team members will be very happy to help!
+
+## Need Help?
+
+If you're stuck or need assistance:
+
+- Check our FAQ section first.
```
Comment (kallewesterling) [inline CONTRIBUTING.md:88]: - [ ] Kalle to add link to the FAQ in the docs here.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+- Reach out on Slack or via email for personalized assistance. (See "Communication Channels" above for links.)
```
Comment (kallewesterling) [inline CONTRIBUTING.md:89]: - [ ] Kalle to change this into "Get in touch" instead. [Not called "Communication Channels" any longer.]

**Review by kallewesterling — COMMENTED** [review]
@tom-andersson left you some question marks here :)

File: CONTRIBUTING.md (hunk 7 above)
Comment (tom-andersson) [reply CONTRIBUTING.md:37]: Agreed.

File: CONTRIBUTING.md (hunk 6 above)
Comment (tom-andersson) [reply CONTRIBUTING.md:36]: Agreed, let's drop this for now.

File: CONTRIBUTING.md (hunk 4 above)
Comment (tom-andersson) [reply CONTRIBUTING.md:28]: Sounds great! You could mention we're planning to restart community calls early in 2024.

File: CONTRIBUTING.md (hunk 1 above)
Comment (tom-andersson) [reply CONTRIBUTING.md:7]: Here's something for now. Tried to keep it to a reasonable length. What do you think?

> Thank you for considering a contribution to DeepSensor! The DeepSensor Python package is motivated by successes in applying convolutional neural processes (ConvNPs) to a range of environmental prediction tasks: sensor placement, forecasting, downscaling, and satellite gap-filling. Behind the success of ConvNPs is the ability to efficiently fuse multi-modal and multi-resolution data, handle missing observations, and capture prediction uncertainty. DeepSensor aims to facilitate further research by plugging together the `xarray`, `pandas`, and `neuralprocesses` packages with a user-friendly interface that enables rapid experimentation. Now, our developers and users form an open-source research community whose vision is to accelerate the next generation of environmental ML research. We are eager to grow DeepSensor from a young Python package into a large and dedicated software community. By joining our efforts, you will be helping to push the frontiers of environmental sciences.

**Comment by kallewesterling** [comment]
All that's pending now is a roadmap, but I think we can merge this PR, and add to the roadmap in the docs when we're ready!

**Pull request merged**